Jumpscale/sal/ubuntu/Ubuntu.py

```diff
@@ -278,66 +347,86 @@ def get_installed_package_names(self):
 
         return self._installed_pkgs
 
-    def is_pkg_installed(self, pkg):
-        return pkg in self._installed_pkgs
+    def apt_find_all(self, package_name):
+        """find all packages match with the package_name
 
-    def apt_find_installed(self, packagename):
-        packagename = packagename.lower().strip().replace("_", "").replace("_", "")
+        :param package_name: ubuntu package name
+        :return: list of package names
+        :rtype: list
+        """
+        package_name = package_name.lower().strip().replace('_', '').replace('_', '')
         if self._cache_ubuntu is None:
             self.apt_init()
         result = []
-        for item in self.get_installed_package_names():
-            item2 = item.replace("_", "").replace("_", "").lower()
-            if item2.find(packagename) != -1:
+        for item in self._cache_ubuntu.keys():
+            if item.replace('_', '').replace('_', '').lower().find(package_name) != -1:
                 result.append(item)
         return result
 
-    def apt_find1_installed(self, packagename):
-        self._logger.info("find 1 package in ubuntu")
-        res = self.apt_find_installed(packagename)
-        if len(res) == 1:
-            return res[0]
-        elif len(res) > 1:
-            raise j.exceptions.RuntimeError("Found more than 1 package for %s" % packagename)
-        raise j.exceptions.RuntimeError("Could not find package %s" % packagename)
+    def is_pkg_installed(self, package_name):
+        """check if the package is installed or not.
+
+        :param package_name: package name
+        :type: str
+        :return: bool
```

Comment:
this should be a description, and there should be `rtype: bool`

Reply:
done